sky/spot/controller.py

```diff
@@ -309,14 +310,25 @@ def start(job_id, task_yaml, retry_until_up):
     except exceptions.SpotUserCancelledError:
         logger.info(f'Cancelling spot job {job_id}...')
         if controller_process is not None:
-            logger.info('Sending SIGTERM to controller process '
+            logger.info('Killing controller process '
```

Comment:
nit: can we move the whole clean up process in the `finally` block into a method `kill_control_process_and_clean_up`, so that we can also call it before `spot_state.set_cancelled()`. The current behavior seems that the spot cluster will be UP for a while even though the status is set to `CANCELLED` already, which may be a bit confusing.

Reply:
good idea